AK/Tests/TestFormat.cpp

```diff
@@ -136,4 +136,15 @@ TEST_CASE(cast_integer_to_character)
     EXPECT_EQ(String::formatted("{:c}", static_cast<unsigned int>('f')), "f");
 }
 
+TEST_CASE(boolean_values)
+{
+    EXPECT_EQ(String::formatted("{}", true), "true");
+    EXPECT_EQ(String::formatted("{}", false), "false");
+    EXPECT_EQ(String::formatted("{:6}", true), "true  ");
+    EXPECT_EQ(String::formatted("{:>4}", false), "false");
+    EXPECT_EQ(String::formatted("{:d}", false), "0");
+    EXPECT_EQ(String::formatted("{:d}", true), "1");
+    EXPECT_EQ(String::formatted("{:#08x}", true), "0x000001");
```

Comment:
Hmm, shouldn't this yield `"0x00000001"`?

Reply:
[This is fmtlib's behavior exactly.][1]

I was thinking about this too, because when formatting numbers one is usually interested in the significant digits and not the total width. But fmtlib does considers the sign and the alternative form prefix in the width calculation.

In my mind it would probably be best to not consider it, because after all we have a "reserved" sign mode which puts a space for positive numbers, where the sign would go otherwise (e.g. `String::formatted("{: }", 1) == " 1"`.) This should be enough to ensure the exact width when required.

I am not completely certain if this is a good idea though, it would differ from how strings are formatted (since they never have any prefixes.)

  [1]: https://godbolt.org/z/noMEfn

Reply:
On a second note, [this is printf's behavior too.][1]

  [1]: https://godbolt.org/z/8n5oMq

Reply:
Huh, well okay :]